src/main/java/com/relayrides/pushy/apns/ApnsConnection.java

```diff
@@ -326,6 +324,10 @@ public void channelInactive(final ChannelHandlerContext context) throws Exceptio
 					this.apnsConnection.listener.handleConnectionClosure(this.apnsConnection);
 				}
 			}
+
+			if (this.apnsConnection.gracefulShutdownTimeoutFuture != null) {
```

Comment:
Very minor: why does this still have `shutdown` when elsewhere that is renamed to `disconnect`?